src/main/java/com/mraof/minestuck/computer/editmode/ClientEditHandler.java

```diff
@@ -128,6 +129,19 @@ private static void addToolTip(ItemStack stack, List<Component> toolTip, GristSe
 			toolTip.add(Component.translatable(GuiUtil.FREE).withStyle(ChatFormatting.GREEN));
 	}
 	
+	@SubscribeEvent
+	public static void onClientTick(TickEvent.ClientTickEvent event)
+	{
+		EditToolDrag.doRecycleCode(event);
+		EditToolDrag.doReviseCode(event);
+	}
+	
+	@SubscribeEvent
+	public static void renderWorld(RenderLevelStageEvent event)
+	{
+		EditToolDrag.renderOutlines(event);
+	}
```

Comment:
I put them there so that all the edit-mode placement-related stuff was at the very least called from inside of ClientEditHandler and ServerEditHandler, but now that I think about it, that _is_ a bit silly. I'll move them.